clouddriver-kubernetes-v2/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/KubernetesV2ProviderSynchronizable.java

```diff
@@ -117,17 +118,38 @@ public void synchronize() {
                 // account exists but has changed
                 changedAccounts.add(managedAccount.getName());
                 newAndChangedAccounts.add(managedAccount.getName());
+              } else {
+                // Current credentials may contain memoized namespaces, we should keep if the
+                // definition has not changed
+                return null;
               }
-
-              accountCredentialsRepository.save(managedAccount.getName(), credentials);
-            });
+              return credentials;
+            })
+        .filter(Objects::nonNull)
+        .forEach(a -> saveToCredentialsRepository(a));
 
     ProviderUtils.unscheduleAndDeregisterAgents(deletedAccounts, catsModule);
     ProviderUtils.unscheduleAndDeregisterAgents(changedAccounts, catsModule);
 
     return newAndChangedAccounts;
   }
 
+  /**
+   * Validate and save credentials to repository
+   *
+   * @param account
+   */
+  protected void saveToCredentialsRepository(KubernetesNamedAccountCredentials account) {
+    // Attempt to get namespaces to resolve any connectivity error without blocking /credentials
+    List<String> namespaces = account.getCredentials().getDeclaredNamespaces();
```

Comment:
One worry here is what happens if the call to get namespaces takes a long time---in that case, I think we'll block the account refresh process while waiting for that network connection.

Is this better than just adding the account, and letting that account's caching thread hang while trying to get namespaces (which wouldn't block other threads)?

Reply:
All we're blocking here is the config refresh not the initial startup of Clouddriver (now in `setup()`). The idea is to do a light validation to make sure adding a bad (or temporarily unavailable) account does not bring Clouddriver down. If the account becomes available later, namespaces will be populated.

Reply:
Ok, I think that makes sense---I *thought* that an unreachable account in clouddriver should only affect the caching thread for that account, which might be better than blocking the whole config refresh thread.  But as this is only testing new accounts that are added in that refresh cycle, the timeout should be fairly quick (and the only effect is that new accounts from that cycle would not be added while we're waiting on an unreachable account).

Reply:
Unfortunately, since the account is added to the account repository, it is used for all operations. New and modified accounts would be affected and potentially blocked but the blocking should only occur once thanks to the memoization you added.

Reply:
Ok, makes sense, thanks for explaining this!